core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -161,8 +160,8 @@ object Pull extends PullLowPriority {
     * Returns a pull that evaluates the supplied by-name each time the pull is used,
     * allowing use of a mutable value in pull computations.
     */
-  def suspend[F[x] >: Pure[x], O, R](p: => Pull[F, O, R]): Pull[F, O, R] =
-    fromFreeC(FreeC.suspend(p.get))
+  def suspend[F[_], O, R](p: => Pull[F, O, R]): Pull[F, O, R] =
```

Comment:
The `F[x] >: Pure[x]` bound is to prevent inference of `Nothing`. Any reason this changes here?

Reply:
Do you have an example code snippet where lack of this `>: Pure[x]` was causing a problem?

In general, I would think that covariance of both `Pull` and `FreeC` would make locally inferring `F[A] = Nothing` not a problem, since it can up-cast it naturally when used in another context.

Reply:
Right, but `Stream` and `Pull` were already covariant in `F`. I can try to come up with a failing example later, but take a look at this blog post which documents why we had to create `Pure` in the first place: https://mpilquist.github.io/blog/2018/07/04/fs2/. Basically, there are cases where inferring `F = Nothing` causes scalac to do the wrong thing.